File: android-app/src/main/java/com/medziku/plantwatering/data/parts/PumpImpl.java
package com.medziku.plantwatering.data.parts;

import android.os.Parcel;
import android.os.Parcelable;

import com.medziku.plantwatering.station.parts.Pump;


public class PumpImpl extends Pump implements Parcelable {

    public PumpImpl(String name, int state, float speed) {
        super(name, state, speed);
    }

    private PumpImpl(Parcel in) {
        super(in.readString(), in.readInt(), in.readFloat());
    }

    public static final Creator<PumpImpl> CREATOR = new Creator<PumpImpl>() {
        @Override
        public PumpImpl createFromParcel(Parcel in) {
            return new PumpImpl(in);
        }

        @Override
        public PumpImpl[] newArray(int size) {
            return new PumpImpl[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(getName());
        dest.writeInt(getState());
        dest.writeFloat(getSpeed());
    }
}

File: android-app/src/main/java/com/medziku/plantwatering/data/parts/PumpImpl.kt
package com.medziku.plantwatering.data.parts

import android.os.Parcel
import android.os.Parcelable

import com.medziku.plantwatering.station.parts.Pump


class PumpImpl : Pump, Parcelable {

    constructor(name: String, state: Int, speed: Float) : super(name, state, speed)

    private constructor(`in`: Parcel) : super(`in`.readString(), `in`.readInt(), `in`.readFloat())

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeInt(state)
        dest.writeFloat(speed)
    }

    companion object {

        val CREATOR: Parcelable.Creator<PumpImpl> = object : Parcelable.Creator<PumpImpl> {
            override fun createFromParcel(`in`: Parcel): PumpImpl {
                return PumpImpl(`in`)
            }

            override fun newArray(size: Int): Array<PumpImpl?> {
                return arrayOfNulls(size)
            }
        }
    }
}
